### PR TITLE
ARROW-15962: [C++][GANDIVA] Fix unhex errors return

### PR DESCRIPTION
**Pull request by ViniciusSouzaRoque**
Modifying function behavior to not pop errors on non-hex inputs

### CODE REVIEW

**Comment by ViniciusSouzaRoque** [comment]
@github-actions crossbow submit java-jars

**Comment by ViniciusSouzaRoque** [comment]
@github-actions crossbow submit java-jars

**Comment by ViniciusSouzaRoque** [comment]
@github-actions crossbow submit java-jars

**Comment by ViniciusSouzaRoque** [comment]
@github-actions crossbow submit java-jars

File: cpp/src/gandiva/precompiled/string_ops.cc [hunk 1]
```diff
@@ -2736,24 +2736,24 @@ const char* to_hex_int32(int64_t context, int32_t data, int32_t* out_len) {
 
 FORCE_INLINE
 const char* from_hex_utf8(int64_t context, const char* text, int32_t text_len,
-                          int32_t* out_len) {
+                          bool text_validity, bool* out_valid, int32_t* out_len) {
```
Comment (vvellanki) [inline cpp/src/gandiva/precompiled/string_ops.cc:2739]: This is now a NULL_INTERNAL function - what should happen if text_validity is false? If the input is null, what should be the output?

I dont see text_validity being used in this function

Comment (ViniciusSouzaRoque) [reply cpp/src/gandiva/precompiled/string_ops.cc:2739]: In this case, i suppose that need returns same a text_len == 0 used below, returning a true for out_valid and one empty string
I go add this on union with that case

Comment (ViniciusSouzaRoque) [reply cpp/src/gandiva/precompiled/string_ops.cc:2739]: correction: in this case i need return false

**Review by vvellanki — APPROVED** [review]

**Pull request closed without merging**